### DISCUSSION TITLE
Having issues with my first floor plan

### DISCUSSION THREAD

**Discussion by martinarva** [ExperienceLovelace/ha-floorplan#313] [General]
Hi!

I made my floorplan and imported it in Inkspace:
<img width="318" alt="Screenshot 2023-01-01 at 14 54 06" src="https://user-images.githubusercontent.com/77044330/210171175-11c920fa-f707-44da-8c9a-c7e216d0bdff.png">

Then draw areas and named objects:
<img width="672" alt="Screenshot 2023-01-01 at 14 56 40" src="https://user-images.githubusercontent.com/77044330/210171224-c16a05cd-c9bf-4630-a896-8e88f0b1ee0b.png">

With these fill settings. Did not use stroke, as there are some overlaping areas an I did not want lines there. Also used some blur, so overlaping areas would be smooth.
<img width="297" alt="Screenshot 2023-01-01 at 14 57 50" src="https://user-images.githubusercontent.com/77044330/210171279-703549bb-2c7b-4ef7-aa8f-8754271f07c0.png">

Here is the svg file: 
![first_floor_v6](https://user-images.githubusercontent.com/77044330/210171295-28e57e41-c44f-4755-8b8c-cec746fa7844.svg)

CSS:
`/* All good :/ */
.entitystate-on {
    opacity: 0;
  }
  
  .entitystate-off {
    opacity: 1;
    fill: rgb(49, 45, 45) !important;
  }`

YAML:
`title: Floorplan
views:
  - theme: Backend-selected
    title: First Floor
    path: floorplan
    panel: true
    badges: []
    cards:
      - config:
          console_log_level: info
          defaults:
            hover_action: hover-info
            tap_action: more-info
          image: /local/floorplan/1st_floor/first_floor_v6.svg
          rules:
            - element: lights_stairs
              entity: light.stair_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_wc
              entity: switch.downstairs_wc_mirror_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_kitchen_ceiling
              entity: light.kitchen_ceiling_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_kitchen_island
              entity: light.kitchen_island_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_sauna_lobby
              entity: light.sauna_lobby_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_sauna_washing_room
              entity: light.sauna_wahing_room_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_sauna
              entity: light.sauna_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_hallway
              entity: light.hallway_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_garage
              entity: light.garage_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_dining_table
              entity: light.dining_table_lights_4
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
            - element: lights_living_room
              entity: light.living_room_lights
              state_action:
                action: call-service
                service: floorplan.class_set
                service_data: entitystate-${entity.state}
              tap_action:
                action: call-service
                service: homeassistant.toggle
          stylesheet: /local/floorplan/1st_floor/first_floor.css
        type: custom:floorplan-card
        full_height: true
`

Light switching works, but i don't get the result where lights on will hide the object.

Lights off result:
<img width="561" alt="Screenshot 2023-01-01 at 15 02 12" src="https://user-images.githubusercontent.com/77044330/210171408-fa110719-fe8a-47ae-a9d0-f7028e1ff139.png">

Lights on (marked on the picture) result:

<img width="555" alt="Screenshot 2023-01-01 at 15 02 43" src="https://user-images.githubusercontent.com/77044330/210171443-0d9b3f4f-5e65-4ba3-9932-03c2b68e329a.png">

So when lights are on, it's even more darker instead of hiding the object.

**Comment by martinarva** [comment]
Looks like I solved it, by adding  !important to on state. :

`.entitystate-on {
    opacity: 0 !important;
  }`


I used the CSS sample from here: https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/blob/main/2_1_Create-your-own-floorplan-in-Inkscape-and-ha-floorplan-logics/home_modern.css

Might be good idea to add the !important to this file as well.

**Reply by exetico** [reply]
I'll definitely change that. Thanks for pointing that out. I guess it didn't conflict with my own draft SVG. 

https://youtu.be/oqO4Vx6U1Mo

Take a look at this, for more information about CSS styling, and the usage of `!important`.

If you're into a bit more advance usage, take a look at:
https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/blob/main/2_2_Create-your-own-floorplan-in-Inkscape-and-ha-floorplan-logics/home_modern_original_scope.css
